test/Chirp.TestE2E/E2ETests.cs

```diff
@@ -695,6 +695,93 @@ public async Task TestShareCheepsVisibilityPrivateTimeline()
+
+        
+        // Clean up and delete data
+        await DeleteUser();   
+    }
+
+    [Test]
+    [Category("End2End")]
+    public async Task CheepingFromPrivateTimeline()
+    {
+        {
+            await RegisterUser();
+            await LoginUser();
+            
+            await _page.GetByRole(AriaRole.Link, new() { Name = "My timeline" }).ClickAsync();
+            await _page.Locator("#CheepText").ClickAsync();
+            await _page.Locator("#CheepText").FillAsync("Hello World!");
+            await _page.GetByRole(AriaRole.Button, new() { Name = "Share" }).ClickAsync();
+            await Expect(_page.Locator("li").Filter(new() { HasText = "Hello World!" }).First).ToBeVisibleAsync();
+
+        
+            // Clean up and delete data
+            await DeleteUser();   
+        }
+    }
+    [Test]
+    [Category("End2End")]
+    public async Task EmptyCheeps()
+    {
+        {
+            await RegisterUser();
+            await LoginUser();
+            
+            await _page.GetByRole(AriaRole.Button, new() { Name = "Share" }).ClickAsync();
+            await Expect(_page.GetByText("At least write something")).ToBeVisibleAsync();
+        
+            // Clean up and delete data
+            await DeleteUser();   
+        }
+    }
+    [Test]
+    [Category("End2End")]
+    public async Task LongCheeps()
+    {
+        {
+            await RegisterUser();
+            await LoginUser();
+            
+            await _page.Locator("#CheepText").ClickAsync();
+            await _page.Locator("#CheepText").FillAsync("Very Long Message Very Long Message Very Long Message Very Long Message Very Long Message Very Long Message Very Long Message Very Long Message Very Long Message Very Long Message Very Long Message Very Long Message Very Long Message Very Long Message Very Long Message Very Long Message Very Long Message Very Long Message ");
+            await _page.GetByRole(AriaRole.Button, new() { Name = "Share" }).ClickAsync();
+            await Expect(_page.GetByText("Maximum length is 160")).ToBeVisibleAsync();
+        
+            // Clean up and delete data
+            await DeleteUser();   
+        }
+    }
+    [Test]
+    [Category("End2End")]
+    public async Task DeletedCheeps()
+    {
+        {
+            await RegisterUser();
+            await LoginUser();
+            
+            await _page.Locator("#CheepText").ClickAsync();
+            await _page.Locator("#CheepText").FillAsync("Hello World!");
+            await _page.GetByRole(AriaRole.Button, new() { Name = "Share" }).ClickAsync();
+            // Clean up and delete data
+            await DeleteUser();   
+            
+            // check that the cheep is deleted
+            await _page.GetByRole(AriaRole.Link, new() { Name = "public timeline" }).ClickAsync();
+            await Expect(_page.GetByText("Public Timeline Jacqualine")).ToBeVisibleAsync();
```

Comment:
It might not be a good idea to depend on other variable factors like database records.
Instead you can use the `.Not.ToBeVisibleAsync()` and maybe check for you message "Hello World!" is **not** visible on the public timeline.

Reply:
I did this since we often seed our database with the default schema, Therfor I belive since we are using this as a prerequisite it is fine to test on

Reply:
I still think this is relevant. We might some day use another seeding file, and if we do that, all of our tests would fail if we relied on that specific file. Its not a hard change to do. You can read more about it [here](https://testsigma.com/blog/assertion-testing/) or even chat about it with chatgpt.